LFS-195: Refactor user dashboard to remove dependency on material dashboard react

OrdiNeu opened the pull request:
[Jira link](https://phenotips.atlassian.net/browse/LFS-195)

This is step 2 in removing `material-dashboard-react` entirely from our codebase, including:
- (From @marta-): a rework of the login page
- (From @marta-): a rework of the header/footer elements
- Removal of the dashboard
- Removal of our MaterialDashboardReact module

OrdiNeu committed:
LFS-195: Refactor user dashboard to remove dependency on material dashboard react

OrdiNeu committed:
LFS-195 / LFS-207: Remove non-functional UI elements

OrdiNeu committed:
[misc] Fix new form button styling

OrdiNeu committed:
[misc] Fix the positioning of the footer using React portals

OrdiNeu committed:
[misc] Fix fonts and font weights throughout application

marta- committed:
[misc] Minor improvements to the New Form button and dialog UI

marta- committed:
LFS-204: Log-in screen redesign
Replaced material-dashbord-react components with Material UI components.
Reverted the look of the Sign in form to be consistent with that of the Sign Up form

marta- committed:
[misc] Fixed markup for login/signup form titles, optimized icon imports, improved code style

sdumitriu reviewed (commented):
JS error on login page:

```
Warning: Failed prop type: ForwardRef(SvgIcon): prop type `component` is invalid; it must be a function, usually from the `prop-types` package, but received `undefined`.
    in ForwardRef(SvgIcon) (created by WithStyles(ForwardRef(SvgIcon)))
    in WithStyles(ForwardRef(SvgIcon)) (created by ForwardRef)
    in ForwardRef
    in ForwardRef (created by SignIn)
    in div (created by ForwardRef(Avatar))
    in ForwardRef(Avatar) (created by WithStyles(ForwardRef(Avatar)))
    in WithStyles(ForwardRef(Avatar)) (created by SignIn)
    in div (created by ForwardRef(Paper))
    in ForwardRef(Paper) (created by WithStyles(ForwardRef(Paper)))
    in WithStyles(ForwardRef(Paper)) (created by SignIn)
    in div (created by SignIn)
    in SignIn (created by WithStyles(SignIn))
    in WithStyles(SignIn) (created by MainLoginContainer)
    in div (created by MainLoginContainer)
    in MainLoginContainer
```

JS warning in vocabulary info dialog:
```
`keepTogether` modifier is required by `arrow` modifier in order to work, be sure to include it before `arrow`!
```

OrdiNeu committed:
LFS-195: Refactor user dashboard to remove dependency on material dashboard react
Update the version of dependencies in login page

OrdiNeu committed:
LFS-195: Refactor user dashboard to remove dependency on material dashboard react
Alter the login page to use the standard header/footer inserts.

OrdiNeu committed:
[misc] Fix warning message about controlled state when vocab checkboxes are selected

OrdiNeu committed:
LFS-195: Refactor user dashboard to remove dependency on material dashboard react
Alter the behaviour of Vocab questions' suggestions list and popper so
that the styles don't break and warnings don't appear.
